core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -669,11 +669,11 @@ private Node stop() {
         injector.getInstance(SnapshotShardsService.class).stop();
         // stop any changes happening as a result of cluster state changes
         injector.getInstance(IndicesClusterStateService.class).stop();
+        injector.getInstance(Discovery.class).stop();
```

Comment:
If the position of this is important I'd leave a comment about why. Even if it is just to stop spammy log messages.